Services/AzureManagementAdapter/AzureManagementAdapter.cs

```diff
@@ -87,6 +79,37 @@ public async Task<MetricsResponseListModel> PostAsync(MetricsRequestListModel re
             return JsonConvert.DeserializeObject<MetricsResponseListModel>(response.Content);
         }
 
+        public async Task CreateOrUpdateVmssAutoscaleSettingsAsync(int vmCount)
+        {
+            await this.CreateOrUpdateAccessTokenAsync();
+
+            var accessToken = $"Bearer {this.ReadSecureString(this.secureAccessToken)}";
+
+            var request = this.PrepareVmssAutoscaleSettingsRequest(accessToken, vmCount.ToString());
+
+            this.log.Debug("Azure Management request content", () => new { request.Content });
+
+            var response = await this.httpClient.PutAsync(request);
+
+            this.log.Debug("Azure management response", () => new { response });
+
+            this.ThrowIfError(response);
```

Comment:
we should check which exception is occurring. For instance, if the user doesn't have enough "cores" left in the subscription, this will keep failing until the user gets more cores, or deletes some VMs. We should make sure that the user knows if this occurs

Reply:
Good suggestion, we should consider adding this as an enhancement. In the current change,  if we will throw exception for any failure.